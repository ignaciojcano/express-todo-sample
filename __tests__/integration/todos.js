const supertest = require('supertest');
const server = require('../../src/server');

const request = supertest(server);

describe('TODO', () => {
  beforeEach(() => {});
  afterEach(() => {});

  it('should fetch all todos', async (done) => {
    const response = await request.get('/todos');
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
    done();
  });

  it('should create a todo', async (done) => {
    const response = await request.post('/todos')
      .send({
        text: 'I just got created',
      });
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
    done();
  });

  it('should fetch a todo', async (done) => {
    const response = await request.get('/todos/1');
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
    done();
  });

  it('should update a todo', async (done) => {
    const response = await request.put('/todos/1')
      .send({
        text: 'A new and updated text',
      });
    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot();
    done();
  });
});
