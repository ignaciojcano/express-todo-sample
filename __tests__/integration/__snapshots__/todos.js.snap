// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TODO should create a todo 1`] = `
Object {
  "message": "implement me!",
}
`;

exports[`TODO should fetch a todo 1`] = `
Object {
  "message": "implement me!",
}
`;

exports[`TODO should fetch all todos 1`] = `
Object {
  "message": "implement me!",
}
`;

exports[`TODO should update a todo 1`] = `
Object {
  "message": "implement me!",
}
`;
